CMakeLists.txt

```diff
@@ -1,23 +1,31 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6 FATAL_ERROR)
 CMAKE_POLICY(VERSION 2.6)
+ENABLE_TESTING()
 
 # This CMake file is used _only_ really to install torchcraft in lua
 IF(LUAROCKS_PREFIX)
     MESSAGE(STATUS "Installing TorchCraft through Luarocks")
     STRING(REGEX REPLACE "(.*)lib/luarocks/rocks.*" "\\1" CMAKE_INSTALL_PREFIX  "${LUAROCKS_PREFIX}")
     MESSAGE(STATUS "Prefix inferred from Luarocks: ${CMAKE_INSTALL_PREFIX}")
 ENDIF()
+
+ADD_SUBDIRECTORY(test)
 FIND_PACKAGE(Torch REQUIRED)
 FIND_PACKAGE(PkgConfig REQUIRED)
 PKG_CHECK_MODULES(ZSTD libzstd>=1.0)
 
-SET(CMAKE_CXX_FLAGS "-std=c++11 -Wall -fPIC -O2")
+SET(CMAKE_CXX_STANDARD 14)
+SET(CMAKE_CXX_STANDARD_REQUIRED ON)
+SET(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -ggdb")
+SET(CMAKE_CXX_FLAGS "-Wall -fPIC -O2")
```

Comment:
Same here, i.e. `SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")`. The optimization levels are best kept in the specific build mode flags, and for `-fPIC` you should use `SET_PROPERTY(TARGET mytarget PROPERTY POSITION_INDEPENDENT_CODE ON)` on a per-target level.